app/main.py

```diff
@@ -2,4 +2,73 @@ class Cargo:
     def __init__(self, weight: int) -> None:
         self.weight = weight
 
-# write your code here
+
+class BaseRobot:
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: list[int] | None = None
+    ) -> None:
+
+        self.name = name
+        self.weight = weight
+        if not coords:
+            coords = [0, 0]
+        self.coords = coords[:2]
+
+    def go_forward(self, step: int = 1) -> None:
+        self.coords[1] += step
+
+    def go_back(self, step: int = 1) -> None:
+        self.coords[1] -= step
+
+    def go_right(self, step: int = 1) -> None:
+        self.coords[0] += step
+
+    def go_left(self, step: int = 1) -> None:
+        self.coords[0] -= step
+
+    def get_info(self) -> str:
+        return f"Robot: {self.name}, Weight: {self.weight}"
+
+
+class FlyingRobot(BaseRobot):
+    def __init__(
+            self,
+            name: str,
+            weight: int,
+            coords: list[int] | None = None
+    ) -> None:
+
+        super().__init__(name, weight, coords)
+        z_coord = 0 if not coords else coords[2]
```

Comment:
```suggestion
        super().__init__(name, weight, coords or [0, 0, 0])
```
z_coord is redundant

Reply:
[0, 0, 0] will be redundant here since the coords are passed to BaseRobot init method where they are truncated anyway. 
But I did get rid of z_coord by condensing the next two lines into one.